src/components/reportPage/UploadBox.jsx

```diff
@@ -2,44 +2,128 @@ import React, { useCallback, useEffect, useState } from "react";
+
 const UploadBox = (props) => {
-  const onDrop = useCallback((files) => {
-    props.setFiles((existingFiles) => [...existingFiles, ...files]);
-  }, []);
+  const onDrop = (files) => {
+    if (files.length) {
+      // Restricts files displayed to fileLimit only
+      if (files.length + currFiles.length > fileLimit) {
+        let updatedFiles = files.slice(0, fileLimit - currFiles.length);
+        setAlert(MAX_FILE_ALERT); //bug: shows up but I don't know why it disappears later
+        props.setFiles((existingFiles) => [...existingFiles, ...updatedFiles]);
+      } else {
+        props.setFiles((existingFiles) => [...existingFiles, ...files]);
+      }
+    }
+  };
+  // didn't use useCallback because the currFiles.length always returned 0 for some reason
+  // const onDrop = useCallback((files) => {
+  //   if (files.length + currFiles.length > fileLimit) {
+  //     let updatedFiles = files.slice(0,fileLimit-currFiles.length);
+  //     props.setFiles((existingFiles) => [...existingFiles, ...updatedFiles]);
+  //     setAlert(MAX_FILE_ALERT);
+  //   }
+  //   else{
+  //     props.setFiles((existingFiles) => [...existingFiles, ...files]);
+  //   }
+
+  // }, []);
 
-  const { setAlert, fileLimit, currFiles } = props;
-  const MAX_FILE_ALERT = `Please upload only up to ${fileLimit} valid files.`;
-  const INVALID_FILETYPE = "You are trying to upload invalid file types.";
+  const { setAlert, fileLimit, currFiles, maxFileSize } = props;
+  const MAX_FILE_ALERT = `Please upload only up to ${props.fileLimit} valid files.`;
+  const INVALID_FILETYPE = `The following file(s) are not of acceptable formats (jpeg, png, mp4):`;
 
   const onDropRejected = (fileRejections) => {
-    const rejectFiles = fileRejections.length;
-    if (rejectFiles) {
-      if (rejectFiles > fileLimit) {
-        setAlert(MAX_FILE_ALERT);
-      } else if (rejectFiles < fileLimit) {
-        // To display the duplicate files uploaded
-        const duplicateFileAlert = fileRejections.map((f) => " " + f.file.name);
-        setAlert("You have already uploaded:" + duplicateFileAlert);
-      } else {
-        setAlert(INVALID_FILETYPE);
+    if (currFiles.length == fileLimit || fileRejections.length > fileLimit) {
+      setAlert(MAX_FILE_ALERT);
+    } else {
+      let errorList = {
+        errorFiles: {
+          overMaxSize: [],
+          duplicateFiles: [],
+          invalidType: [],
+        },
+        errorMsgs: {
+          overMaxSize: `The following file(s) are larger than ${maxFileSize} MB: `,
+          duplicateFiles: "The following file(s) have already been uploaded:",
+          invalidType: INVALID_FILETYPE,
+        },
+      };
+      let errorFiles = errorList.errorFiles;
+      for (let f in fileRejections) {
+        const { errors, file } = fileRejections[f];
+        let errorType = errors[0].code;
+        if (errorType === "file-invalid-type") {
+          errorFiles.invalidType = [...errorFiles.invalidType, file.name];
```

Comment:
You should use `errorFiles.invalidType.push(file.name)`
We only need to use `= [...errorFiles.invalidType, file.name]` when we are setting an array defined using the `useState` hook